test/plug_test.exs

```diff
@@ -18,10 +20,17 @@ defmodule Repo do
   def get(Post, 2), do: %Post{id: 2, user_id: 2 }
   def get(Post, _), do: nil
 
-  def all(_), do: [%Post{id: 1}, %Post{id: 2}]
+  def all(_), do: [%Post{id: 1}, %Post{id: 2, user_id: 2}]
+
+  def preload(%Post{id: 1}, :user), do: %Post{id: 1}
+  def preload(%Post{id: 2, user_id: 2}, :user), do: %Post{id: 2, user_id: 2, user: %User{id: 2}}
+  def preload([%Post{id: 1},  %Post{id: 2, user_id: 2}], :user), do: [%Post{id: 1}, %Post{id: 2, user_id: 2, user: %User{id: 2}}]
+  def preload(resources, _), do: resources
+
```

Comment:
Please remove the empty line here for consistency with the rest of the project :smiley: